client/funq/client.py

```
@@ -170,6 +170,49 @@ def quit(self):
         """
         self._raw_send('quit', {})
 
+    def action(self, alias=None, path=None, timeout=10.0,
+               timeout_interval=0.1, wait_active=True):
+        """
+        Returns an instance of a :class:`funq.models.Action` or derived
+        identified with an alias or with its complete path.
+
+        Example::
+
+          action = client.action('my_alias')
+
+        :param alias: alias defined in the aliases file.
+        :param path: complete path for the action
+        :param timeout: if > 0, tries to get the action until timeout
+                        is reached (second)
+        :param timeout_interval: time between two atempts to get an action
+                                 (seconds)
+        :param wait_active: If true - the default -, wait until the action
+                            become visible and enabled.
+        """
+        if not (alias or path):
+            raise TypeError("alias or path must be defined")
+
+        if alias:
+            path = self.aliases[alias]
+
+        wdata = [None]
+
+        def get_action():
+            """ Try to get the action """
+            try:
+                wdata[0] = self.send_command('widget_by_path', path=path)
+                return True
+            except FunqError as err:
+                if err.classname != 'InvalidWidgetPath':
+                    raise
```

Maybe raise a more specific Exception here, something like `InvalidWidgetPathError`

---

This is the same pattern used everywhere in this file. Though I can't remember why, we may want to look into this (and remove the pattern's duplication) later.